### PR TITLE
GH-44178: [GLib][Ruby][Flight] Allow setting CallOption timeout 

### PR DESCRIPTION
**Pull request by datbth**
### Are there any user-facing changes?

* GitHub Issue: #44178

### CODE REVIEW

File: c_glib/arrow-flight-glib/client.cpp [hunk 1]
```diff
@@ -294,6 +294,23 @@ gaflight_call_options_foreach_header(GAFlightCallOptions *options,
   }
 }
 
+/**
+ * gaflight_call_options_set_timeout:
+ * @options: A #GAFlightCallOptions.
+ * @value: The timeout value (in seconds).
```
Comment (kou) [inline c_glib/arrow-flight-glib/client.cpp:300]:
```suggestion
 * @timeout: The timeout in seconds.
```

File: c_glib/arrow-flight-glib/client.cpp [hunk 2]
```diff
@@ -294,6 +294,23 @@ gaflight_call_options_foreach_header(GAFlightCallOptions *options,
...
+ *
+ * Set the timeout (in seconds).
```
Comment (kou) [inline c_glib/arrow-flight-glib/client.cpp:302]:
```suggestion
 * Set the timeout in seconds.
 *
 * An optional timeout for this call. Negative durations
 * mean an implementation-defined default behavior will be used
 * instead. This is the default value.
```

File: c_glib/arrow-flight-glib/client.cpp [hunk 3]
```diff
@@ -294,6 +294,23 @@ gaflight_call_options_foreach_header(GAFlightCallOptions *options,
...
+ *
+ * Since: 18.0.0
+ */
+void
+gaflight_call_options_set_timeout(GAFlightCallOptions *options,
+                                 const gint64 *value)
```
Comment (kou) [inline c_glib/arrow-flight-glib/client.cpp:308]:
```suggestion
                                  gdouble timeout)
```

File: c_glib/arrow-flight-glib/client.cpp [hunk 4]
```diff
@@ -294,6 +294,23 @@ gaflight_call_options_foreach_header(GAFlightCallOptions *options,
...
+{
+  auto flight_options = gaflight_call_options_get_raw(options);
+  flight_options->timeout = arrow::flight::TimeoutDuration(*value);
```
Comment (kou) [inline c_glib/arrow-flight-glib/client.cpp:311]:
```suggestion
  flight_options->timeout = arrow::flight::TimeoutDuration(timeout);
```

**Review by kou — COMMENTED** [review]
This approach is acceptable but implementing this as a property is better.
If you aren't familiar with GObject's property, can I work on this?

**Comment by datbth** [comment]
Many thanks for commenting, @kou.  

Yes please work on this per your convenience. Thank you!

Indeed I'm not familiar with GObject introspection.  
Besides, I'm not a C++ developer so I'm actually still have troubles setting up my dev environment (currently around building/linking the local build of arrow 18). I wonder if there is a Slack/Discord where I can bother and learn more from you guys lol.  
I'm only trying out in my spare time so it very likely will take much longer for me to complete this PR.

**Comment by kou** [comment]
OK. I'll open a PR for this later.

How about using https://ursalabs.zulipchat.com/ ?

**Comment by kou** [comment]
GH-44181

**Pull request closed without merging**